packages/commonwealth/client/scripts/views/pages/Editor/Editor.tsx

```diff
@@ -0,0 +1,262 @@
+  const imageUploadHandlerDelegateLocal = useImageUploadHandlerLocal();
+  const imageUploadHandlerDelegateS3 = useImageUploadHandlerS3();
+
+  return useCallback(
+    async (file: File): Promise<ImageURL> => {
+      switch (imageHandler) {
+        case 'S3':
+          return await imageUploadHandlerDelegateS3(file);
+        case 'local':
+          return await imageUploadHandlerDelegateLocal(file);
+      }
+
+      throw new Error('Unknown image handler: ' + imageHandler);
+    },
+    [
+      imageHandler,
+      imageUploadHandlerDelegateLocal,
+      imageUploadHandlerDelegateS3,
+    ],
+  );
+}
+
+export const Editor = memo(function Editor(props: EditorProps) {
+  const errorHandler = useEditorErrorHandler();
+  const [dragging, setDragging] = useState(false);
+  const [uploading, setUploading] = useState(false);
+
+  const dragCounterRef = useRef(0);
+
+  const mode = props.mode ?? 'desktop';
+  const imageHandler: ImageHandler = props.imageHandler ?? 'S3';
+
+  const placeholder = props.placeholder ?? 'Share your thoughts...';
+
+  const mdxEditorRef = React.useRef<MDXEditorMethods>(null);
+
+  const imageUploadHandlerDelegate = useImageHandler(imageHandler);
+
+  const imageUploadHandler = useCallback(
+    async (file: File) => {
+      try {
+        setUploading(true);
+        return await imageUploadHandlerDelegate(file);
+      } finally {
+        setUploading(false);
+      }
+    },
+    [imageUploadHandlerDelegate],
+  );
+
+  const handleFile = useCallback(async (file: File) => {
+    const text = await fileToText(file);
+    mdxEditorRef.current?.setMarkdown(text);
+  }, []);
+
+  const handleImportMarkdown = useCallback(
+    (file: File) => {
+      async function doAsync() {
+        await handleFile(file);
+      }
+
+      doAsync().catch(console.error);
+    },
+    [handleFile],
+  );
+
+  const handleDropAsync = useCallback(
+    async (event: React.DragEvent) => {
+      console.log(event.dataTransfer.files.length);
+
+      const nrFiles = event.dataTransfer.files.length;
+
+      try {
+        if (nrFiles === 1) {
+          const type = event.dataTransfer.files[0].type;
+
+          if (['text/markdown', 'text/plain'].includes(type)) {
+            await handleFile(event.dataTransfer.files[0]);
+          } else {
+            // TODO: use a snackbar
+            console.log('File not markdown');
+          }
+        }
+
+        if (nrFiles <= 0) {
+          // TODO: use a snackbar
+          console.log('No files given');
+          return;
+        }
+
+        if (nrFiles > 1) {
+          // TODO: use a snackbar
+          console.log('Too many files given');
+          return;
+        }
+      } finally {
+        setDragging(false);
+      }
+    },
+    [handleFile],
+  );
+
+  // TODO: handle html files but I'm not sure about the correct way to handle it
+  // because I have to convert to markdown.  This isn't really a typical use
+  // case though
```

Comment:
do we want to support html -> markdown dropped file conversion per editor specs?

Reply:
Only *paste* of HTML, into the editor. NOT a raw HTML file but I think you might have given me an idea on how to handle a raw HTML file but I don't think this actually going to be used.